### ISSUE TITLE
simulator crashes with HTTP sync turned on

### ISSUE THREAD

**Issue by yonahforst** [transistorsoft/react-native-background-geolocation#89]
Simulator recently started crashing when I have the HTTP sync enabled. Doesn't crash when running on the device. Not sure what other info i can provide. Here's a screenshot of the debugger. Is there a way to export crash reports?

![screen shot 2016-06-10 at 10 29 38](https://cloud.githubusercontent.com/assets/1440796/15958954/a80ac4b0-2ef8-11e6-9bad-3f48c0e5ebd2.png)

**Comment by yonahforst** [comment]
@christocracy  still getting this issue. Is there anything I can do to help debug?

**Comment by christocracy** [comment]
Plugin version?

The problem is likely caused by something weird you're doing in your code.
Show me your code.

On Monday, July 4, 2016, Yonah Forst <notifications@github.com> wrote:

> @christocracy <https://github.com/christocracy> still getting this issue.
> Is there anything I can do to help debug?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/transistorsoft/react-native-background-geolocation/issues/89#issuecomment-230262412>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAA6ly9W6aTd1UVpqswKqYXwXgVe5Oq6ks5qSOSagaJpZM4Iyvjr>
> .
>


-- 
Snet form Gmail Mobile

**Comment by yonahforst** [comment]
sorry, should have included that to begin with: 0.6.1

here's the code:

```js
//StartupTasks.js

const userId = '******'
const password = '******'
const basic_auth = 'Basic ' + base64.encode(userId + ':' + password)

class StartupTasks {
  initialize(store) {
    this._store = store
  }

  _getMyId() {
    return this._store.getState().me.id
  }

  startBackgroundLocationMonitoring() {
    BackgroundGeolocation.configure({
      desiredAccuracy: 100,
      distanceFilter: 10,
      stopTimeout: 2, // 2 minutes
      debug: false,
      activityType: 'Fitness',
      locationAuthorizationRequest: 'Always',
      locationUpdateInterval: 10000,
      fastestLocationUpdateInterval: 10000,
			// stopOnTerminate: false,
			// useSignificantChangesOnly: true,
      // HTTP / SQLite config
      url: `https://apiv1.proximator.info:8443/user/${this._getMyId()}`,
      headers: {
        'Authorization': basic_auth,
        'Accept': 'application/json',
      },
      maxDaysToPersist: 0,    // <-- Maximum days to persist a location in plugin's SQLite database when HTTP fails
      method: 'POST'
    });
    BackgroundGeolocation.on('location', loc => {
      this._store.dispatch(updateMyUser({location: loc}))
    });
    BackgroundGeolocation.on('error', error => console.log(error));
    BackgroundGeolocation.start();
  }

  subscribeToPushNotifications() {
    PushNotifications.initialize(this._getMyId(), note => {
      this.updateLocationAndPushToServer()
    })
  }

  updateLocationAndPushToServer() {
    // Fetch current position
    BackgroundGeolocation.getCurrentPosition({timeout: 30}, location => {
      this._store.dispatch(updateMyUser({location: location}))
      fetch(`https://apiv1.proximator.info:8443/user/${this._getMyId()}`, {
        method: 'POST',
        headers: new Headers({
          'Authorization': basic_auth,
          'Content-Type': 'application/json',
        }),
        body: JSON.stringify({location: location, updatedAt: location.timestamp}),
      }).then(res => {
        if (res.status != 201) {
          console.log('background fetch: send location to server failed:', res)
        }
      }).catch(e => console.log('background fetch: couldnt update location: ', e))

    }, function(error) {
      console.log("Location error: " + error);
    })
  }

}


//root.js
//...
export default class Root extends Component {
	state = {}
	async componentWillMount() {
		let store = await createStore()
		this.setState({store: store})

		StartupTasks.initialize(store)
		StartupTasks.startBackgroundLocationMonitoring()
		//...
	}
 //...
}

**Comment by christocracy** [comment]
I've refactored a bunch of stuff related to "background task-management" in latest version.  This problem should be fixed now.
